pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldby"`
+	// How long the lock is valid.
+	LeaseTime uint64 `json:"duration"`
+	// When the lock was first acquired.
+	AcquiredTime string `json:"atime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime string `json:"rtime,omitempty"`
+}
+
+// Lock represents the configuration of a resource lock.
+type Lock struct {
```

Comment:
My argument is that a traditional lock guarantees exclusion until it is explicitly released. This is distinct from a lease which guarantees exclusion for a specific duration. The design doc also uses lease not lock. I agree that what we have here could be considered both a lease since it expires and a lock since it can be explicitly released, although the common case right now is expiration. I also agree that it's not a big deal.